src/lib/io/include/sol-uart.h

```diff
@@ -53,36 +53,47 @@ extern "C" {
 
 struct sol_uart;
 
-struct sol_uart *sol_uart_open(const char *port_name);
+enum sol_uart_speed {
+    SOL_UART_SPEED_9600 = 0,
+    SOL_UART_SPEED_19200,
+    SOL_UART_SPEED_38400,
+    SOL_UART_SPEED_57600,
+    SOL_UART_SPEED_115200
+};
+
+enum sol_uart_data_bits {
+    SOL_UART_DATA_BITS_8 = 0,
+    SOL_UART_DATA_BITS_7,
+    SOL_UART_DATA_BITS_6,
+    SOL_UART_DATA_BITS_5
+};
+
+enum sol_uart_parity {
+  SOL_UART_PARITY_DISABLE = 0,
+  SOL_UART_PARITY_ENABLE_EVEN,
+  SOL_UART_PARITY_ENABLE_ODD
+};
+
+enum sol_uart_stop_bits {
```

Comment:
I dislike functions with many parameters, since it's too easy to mix their orders (hell, I ended up mixing the order of `memset()` and took me a while to find the bug because the compiler didn't complain!). Parameters being enums will help here, though, so whatever floats @zehortigoza's boat.